### ISSUE TITLE
Tutorial - Moves are depleted, but can't end turn.

### ISSUE THREAD

**Issue by Konrad22** [wesnoth/wesnoth#3039]
I was just browsing the steam forum and someone complained about this. They managed to find the cause, the player moved to a village when he was supposed to move to the castle. The 'problem' is, there is no message if you move to the wrong place. Noone tells you to undo the turn with 'u'. If you are new to this game, you'd maybe be stuck there for a bit and get really mad, because you have no idea what's going on. :/

**Comment by Konrad22** [comment]
I agree.

**Comment by sevu** [comment]
This one isn't part of #1584, eventually one could group it under point 1a)

I could imagine that there are a few more situations where an undo reminder is needed, please tell if you encounter some.

**Comment by Konrad22** [comment]
Well...literally any situation when you are told to go 'somewhere' and you do something that leads you to having not enough mp to go 'somewhere'.

**Comment by msrinteractive** [comment]
Would it be possible to implement a simple button to restart the tutorial? Perhaps have it float somewhere over the bottom of the map, or pop it into an unused area in the right sidebar?

**Comment by CelticMinstrel** [comment]
Hmm, that's not a bad idea but I wonder if it's actually possible to add theme buttons from Lua… maybe? And if that's possible, the easiest way to restart is to probably end the scenario and set the next scenario to the same one.

**Comment by Ferk** [comment]
Ideally, there should be some event that can be triggered when a side has no more moves left. Then at that point the tutorial can check if the player is in the right place and ask him to undo if he's not.

Also, it would be a nice feature in general if that event was by default highlighting the "End turn" button when all units finished their moves, it'd be a convenient visual feedback that saves you from always pressing "n" to be sure.

**Comment by Ferk** [comment]
Also, maybe the tutorial could use labels to indicate the tiles that are villages and the tile that's the keep.

**Comment by bandita137** [comment]
The "Movement" step right before tells us how to undo. So I think it is not a bug, rather too much info unread in case someone misses this option.

>  To move, simply click on Konrad and then your destination. (You can press <b>u</b> to undo a move if you move to the wrong place.)

**Comment by Konrad22** [comment]
I don't think we ever said it is a bug? (even though it has this really neat label)
It's more about adding this specific info (press u to undo) whenever you happen to move wrong. So far you are only told once about undoing movements, and it's in a wall of text. The tutorial is for new players, who are a bit screwed if they missed that info. And that's not good. :/

**Comment by bandita137** [comment]
You're right. Spent some time trying to set up an event+filter to check the 0 movement points but it was over my current level.

**Comment by Ferk** [comment]
An alternative would be to modify the engine so a parameter for 'disallow_end_turn" can be given to change the text that shows when the "end turn" button is pressed.

Disallowing the end of turn changes a fundamental game and UI mechanic that deserves clear explanation, not only for the tutorial but for any scenario that uses it, IMHO.

**Comment by bandita137** [comment]
With my most recent version of the fix the undo message will not pop up if the player targets the quintain and gets too far away from the designated target, e.g. the villages (depending on current turn) yet near enough to attack the quintain. (This is in line with allowing end turn after attacking the quintain instead of occupying a village as instructed.)
